src/test/java/org/jvnet/hudson/test/JenkinsRuleTest.java

```diff
@@ -127,6 +142,206 @@ public void testTokenHelperMethods() throws Exception {
         makeRequestAndAssertLogin(wc, "charlotte");
     }
 
+    @Test
+    public void getJSONTests() throws Exception {
+
+        // Testing a simple GET that should answer 200 OK and a json
+        JenkinsRule.JSONWebResponse response = j.getJSON("testing-cli/getMyJsonObject");
+        assertTrue(response.getContentAsString().contains("I am JenkinsRule"));
+        assertEquals(response.getStatusCode(), 200);
+
+        // Testing a simple GET with parameter that should answer 200 OK and a json
+        response = j.getJSON("testing-cli/getWithParameters?paramValue=whatelse");
+        assertTrue(response.getContentAsString().contains("I am JenkinsRule whatelse"));
+        assertEquals(response.getStatusCode(), 200);
+
+        //Testing with a GET that the test expect to raise an server error: we want to be able to assert the status
+        JenkinsRule.WebClient webClientAcceptException = j.createWebClient();
+        webClientAcceptException.setThrowExceptionOnFailingStatusCode(false);
+        response = j.getJSON("testing-cli/getError500", webClientAcceptException);
```

Comment:
For example https://github.com/jenkinsci/jenkins-test-harness/pull/378#discussion_r789919555 would mean that this could be simplified to

```suggestion
        response = j.createWebClient().withThrowExceptionOnFailingStatusCode(false).getJSON("testing-cli/getError500");
```